Nodejs/Product/Nodejs/NpmUI/NpmWorker.cs

```diff
@@ -106,74 +106,135 @@ private void Run()
+                            "email": "doug@somethingdoug.com"
+                          }
+                        ]
+                      }*/
+                    var reader = new StreamReader(response.GetResponseStream());
+                    using (var jsonReader = new JsonTextReader(reader))
+                    {
+                        while (jsonReader.Read())
                         {
-                          "package": {
-                            "name": "express",
-                            "scope": "unscoped",
-                            "version": "4.15.2",
-                            "description": "Fast, unopinionated, minimalist web framework",
-                            "keywords": [ "express", "framework", "sinatra", "web", "rest", "restful", "router", "app", "api" ],
-                            "date": "2017-03-06T13:42:44.853Z",
-                            "links": {
-                              "npm": "https://www.npmjs.com/package/express",
-                              "homepage": "http://expressjs.com/",
-                              "repository": "https://github.com/expressjs/express",
-                              "bugs": "https://github.com/expressjs/express/issues"
-                            },
-                            "author": {
-                              "name": "TJ Holowaychuk",
-                              "email": "tj@vision-media.ca"
-                            },
-                            "publisher": {
-                              "username": "dougwilson",
-                              "email": "doug@somethingdoug.com"
-                            },
-                            "maintainers": [
-                              {
-                                "username": "dougwilson",
-                                "email": "doug@somethingdoug.com"
-                              }
-                            ]
-                          },
-                          "score": {
-                            "final": 0.9549640105248649,
-                            "detail": {
-                              "quality": 0.9427473299991661,
-                              "popularity": 0.9496544159654299,
-                              "maintenance": 0.9707450455348992
+                            switch (jsonReader.TokenType)
+                            {
+                                case JsonToken.StartObject:
+                                    var token = JToken.ReadFrom(jsonReader);
+                                    var package = ReadPackage(token, new NodeModuleBuilder());
+                                    if (package != null)
+                                    {
+                                        return new[] { package };
+                                    }
+                                    break;
+                                default:
+                                    throw new InvalidOperationException("Unexpected json token.");
```

Comment:
Would it be interesting to know something about the bad token?

Reply:
yes